basics/indexing/forward-index.md

```diff
@@ -31,10 +45,24 @@ When a column is physically sorted, Pinot uses a sorted forward index with run-l
 
 The Sorted forward index has the advantages of both good compression and data locality. The Sorted forward index can also be used as an inverted index.
 
-### Real-time tables
+Sorted forward index are active when:
+* The segment is sorted by this column
+* The dictionary is enabled for that column (see [dictionary documentation page](dictionary-index.md)).
+
+If you are ingesting multiple segments you will need to make sure that data is sorted within each segment - you don't need to sort the data across segments.
```

Comment:
```suggestion
If you are ingesting multiple segments, make sure that data is sorted within each segment, so you don't need to sort the data across segments.
```

Reply:
The rephrase changes the meaning of the text.

What I want to say is:
1. The user is responsible to sort rows in the same segment. That means that for all two rows `r1` and `r2`, if `r1` position is lower than `r2`, then the value of the indexed column for `r1` must be <= than the value of that column for `r2`. A violation in this invariant will produce a segment where the column is not sorted.
2. Each segment is independent of each other. Therefore there is no invariant between rows of different segments. For example, if `r1` belongs to segment `s1` and `r21` and `r22` belong to another segment `s2`, then `r1` may be smaller than `r21` and larger than `r22`. There is no order restriction between rows in different segments.

Therefore you need to make sure that data is sorted within each segment *but* you don't need to sort the data across segments. The first condition (sorting rows within a segment) has nothing to do with the second (to require or not to sort data across segments). Therefore I don't think `so` should be used here.

Reply:
> Make sure that data is sorted within each segment but you don't need to sort the data across segments. The first condition (sorting rows within a segment) has nothing to do with the second (to require or not to sort data across segments). Therefore I don't think so should be used here.

Remove "so" to be sure these conditions aren't linked. The first sentence is clear. For the second sentence, "You don't need to sort..." it'd be helpful to consider:
- Although they don't need to sort across segments, is it ok if they do? If the answer's no, we could clarify: In this case, we do not recommend sorting across segments.
- If they can choose to sort data across segments or not, we could say: "Although you may choose to sort data across segments, it's not required."